lib/connections/quic_listen.go

```diff
@@ -59,7 +59,8 @@ func (t *quicListener) OnNATTypeChanged(natType stun.NATType) {
 func (t *quicListener) OnExternalAddressChanged(address *stun.Host, via string) {
 	var uri *url.URL
 	if address != nil {
-		uri = &(*t.uri)
+		uri = &url.URL{}
```

Comment:
I'd have preferred this as

```
copy := *t.uri
uri = &copy
```

to be more explicit about it being a copy that needs to happen, but /shrug